broker/submissions/export_to_spreadsheet_service.py

```diff
@@ -33,23 +33,53 @@ def export_and_save(self, submission_uuid: str, storage_dir: str):
             submission_url = submission['_links']['self']['href']
         except Exception as e:
             self.logger.error(e)
-            raise Exception(f'An error occurred in retrieving the submission with uuid {submission_uuid}: {str(e)}')
-
-        create_date = datetime.now(timezone.utc)
-
-        self._patch(submission_url, create_date)
+            raise Exception(f'An error occurred in retrieving the submission with uuid {submission_uuid}: {str(e)}') from e
 
+        create_date = self.update_spreadsheet_start(submission_url)
         spreadsheet_details = self.get_spreadsheet_details(create_date, storage_dir, submission_uuid)
-
         workbook = self.export(submission_uuid)
+        self.save_spreadsheet(spreadsheet_details, workbook)
+        self.link_spreadsheet(spreadsheet_details, submission_uuid)
+        self.update_spreadsheet_finish(create_date, submission_url)
+        self.copy_to_s3_staging_area(spreadsheet_details, submission_uuid)
+        self.logger.info(f'Done exporting spreadsheet for submission {submission_uuid}!')
 
+    def save_spreadsheet(self, spreadsheet_details, workbook):
```

Comment:
can be made static

Reply:
This is not urgent. This is a cleanup task. Let’s get the functionality working end to end.